pkg/network/ovn_kubernetes.go

```diff
@@ -194,28 +198,40 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
-// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
-// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
-func GetGatewayMode(kubeClient client.Client) (string, error) {
-	defaultGatewayMode := "shared"
+// bootstrapOVNConfig returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns default configuration for OCP clusters using OVN-Kubernetes
+func bootstrapOVNConfig(kubeClient client.Client) (*bootstrap.OVNConfigBoostrapResult, error) {
+	ovnConfigResult := &bootstrap.OVNConfigBoostrapResult{
+		GatewayMode:            OVN_SHARED_GW_MODE,
+		EnableEgressIP:         true,
+		DisableSNATMutlipleGWs: false,
+	}
+
 	cm := &corev1.ConfigMap{}
 	nsn := types.NamespacedName{Namespace: "openshift-network-operator", Name: "gateway-mode-config"}
 	err := kubeClient.Get(context.TODO(), nsn, cm)
 
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			klog.Warningf("Did not find gateway-mode-config. Using default mode: \"%s\"", defaultGatewayMode)
-			return defaultGatewayMode, nil
+			klog.Infof("Did not find gateway-mode-config. Using default OVN configuration: %+v", ovnConfigResult)
+			return ovnConfigResult, nil
 		} else {
-			return "", fmt.Errorf("Could not determine gateway mode: %w", err)
+			return nil, fmt.Errorf("Could not determine gateway mode: %w", err)
 		}
 	}
-	if cm.Data["mode"] != "shared" && cm.Data["mode"] != "local" {
-		klog.Warningf("Ignoring gateway-mode-config %s. Does not match \"shared\" or \"local\"", cm.Data["mode"])
-		return defaultGatewayMode, nil
+	modeOverride := cm.Data["mode"]
+	_, disableSNATMultipleGWsOverride := cm.Data["disable-snat-multiple-gws"]
+	if modeOverride != OVN_SHARED_GW_MODE && modeOverride != OVN_LOCAL_GW_MODE {
+		klog.Warningf("gateway-mode-config does not match %q or %q, is: %q. Using default OVN configuration: %+v", OVN_LOCAL_GW_MODE, OVN_SHARED_GW_MODE, modeOverride, ovnConfigResult)
+		return ovnConfigResult, nil
+	}
+	ovnConfigResult.GatewayMode = modeOverride
+	if disableSNATMultipleGWsOverride {
```

Comment:
disableSNATMultipleGWs with local gateway mode doesn't do anything, it is assumed to always be enabled for that mode. That being said I dont think it hurts to pass it as enabled to ovnkube with local gw mode +  I dont see why anyone would set this with local gateway mode so safe to ignore checking gateway mode here.